src/Init/Data/List/Nat/Count.lean

```diff
@@ -28,4 +28,59 @@ theorem count_set [BEq α] (a b : α) (l : List α) (i : Nat) (h : i < l.length)
     (l.set i a).count b = l.count b - (if l[i] == b then 1 else 0) + (if a == b then 1 else 0) := by
   simp [count_eq_countP, countP_set, h]
 
+/--
+The number of elements satisfying a predicate in a sublist is at least the number of elements satisfying the predicate in the list,
+minus the difference in the lengths.
+-/
+theorem Sublist.le_countP (s : l₁ <+ l₂) (p) : countP p l₂ - (l₂.length - l₁.length) ≤ countP p l₁ := by
```

Comment:
Maybe avoiding `Nat.sub`, since that is the general advise, isn’t it?
```suggestion
theorem Sublist.le_countP (s : l₁ <+ l₂) (p) : countP p l₂ + l₁.length ≤ countP p l₁ + l₂.length := by
```
(This is based on a suggestion by @jcommelin on a somewhat similar lemma of mine in https://github.com/leanprover-community/mathlib4/pull/15383#discussion_r1718761193)

But I agree that reads much less natural that way.

Reply:
Yeah, I trust in `omega` for things like this.